src/app/modules/library/personal-library/personal-library.component.ts

```diff
@@ -80,6 +117,69 @@ export class PersonalLibraryComponent extends LibraryComponent {
   protected getDetailsComponent(): any {
     return PersonalLibraryDetailsComponent;
   }
+
+  public filterUpdated(filterValues: ProjectFilterValues = null): void {
+    super.filterUpdated(filterValues);
+    this.filteredProjects = this.filteredProjects.filter(
+      (project) => project.tags.includes('archived') == this.showArchived
+    );
+  }
+
+  protected switchActiveArchivedView(): void {
+    this.filterUpdated();
+    this.clearSelectedProjects();
+  }
+
+  pageChange(event?: PageEvent, scroll?: boolean): void {
+    super.pageChange(event, scroll);
+    this.clearSelectedProjects();
+  }
+
+  protected projectSelectionChanged(event: any): void {
+    if (event.checked) {
+      this.selectedProjects.update((selectedProjects) => {
+        selectedProjects.push(event.project);
+        return selectedProjects;
+      });
+    } else {
+      this.selectedProjects.update((selectedProjects) => {
+        selectedProjects.splice(selectedProjects.indexOf(event.project), 1);
+        return selectedProjects;
+      });
+    }
+  }
+
+  protected archiveSelectedProjects(archive: boolean): Subscription {
+    return this.archiveProjectService[archive ? 'archiveProjects' : 'unarchiveProjects'](
+      this.selectedProjects()
+    ).subscribe({
+      next: () => {
+        this.archiveProjectService.showArchiveProjectsSuccessMessage(
+          this.selectedProjects(),
+          archive
+        );
+        this.archiveProjectService.updateProjectsArchivedStatus(this.selectedProjects(), archive);
+        this.clearSelectedProjects();
+      },
+      error: () => {
+        this.archiveProjectService.showArchiveProjectErrorMessage(archive);
+      }
+    });
+  }
+
+  private clearSelectedProjects(): void {
+    this.selectAllProjectsChanged(false);
+  }
+
+  protected selectAllProjectsChanged(selectAll: any): void {
```

Comment:
Make function name active (instead of reactive, "...Changed"), and add boolean type?
```suggestion
  protected selectAllProjects(selectAll: boolean): void {
```